packages/mcfunction/src/parser/macro.ts

```diff
@@ -0,0 +1,123 @@
+import * as core from '@spyglassmc/core'
+import { localeQuote, localize } from '@spyglassmc/locales'
+import type { MacroNode } from '../node/index.js'
+
+/**
+ * Parse a macro line.
+ */
+export function macro(): core.Parser<MacroNode> {
+	return (
+		src: core.Source,
+		ctx: core.ParserContext,
+	): core.Result<MacroNode> => {
+		const ans: MacroNode = {
+			type: 'mcfunction:macro',
+			range: core.Range.create(src.cursor),
+			children: [],
+		}
+
+		// Skip the starting '$'
+		src.skip()
+
+		let start = src.cursor
+		let hasMacroArgs = false
+
+		while (src.canReadInLine()) {
+			src.skipUntilOrEnd(core.LF, core.CR, '$')
+			if (src.peek(2) === '$(') {
+				hasMacroArgs = true
+				// Add the other stuff before this macro key
+				const other = src.sliceToCursor(start)
+				if (other.length > 0) {
+					ans.children.push({
+						type: 'mcfunction:macro/other',
+						range: core.Range.create(start, src.cursor),
+						value: other,
+					})
+					start = src.cursor
+				}
+				// Parse the macro key
+				const key = validateMacroArgument(src, ctx, start)
+				if (key === core.Failure) {
+					return core.Failure
+				}
```

Comment:
i think these early returns are a breaking change. i reverted the changes to `parser/macro.ts` found in [`331919e` (#1196)](https://github.com/SpyglassMC/Spyglass/pull/1196/commits/331919ec14bb6fff77bf852598f2dc8dc4775ad8#diff-de781f026023e7c4048802b049dce94c3eff97dba7987568181956055fe0ce28), and i was able to run the server.

tests also still pass fine. not sure what the initial motivation was behind this? i'm not getting any TS errors when i remove those changes